zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -411,6 +411,87 @@ impl<P: consensus::Parameters> RusqliteMigration for WalletMigration2<P> {
     }
 }
 
+struct WalletMigrationAddTxViews;
+
+migration!(
+    WalletMigrationAddTxViews,
+    "282fad2e-8372-4ca0-8bed-71821320909f",
+    ["be57ef3b-388e-42ea-97e2-678dafcf9754"],
+    "Add views over transaction & note data."
+);
+
+impl RusqliteMigration for WalletMigrationAddTxViews {
+    type Error = WalletMigrationError;
+
+    fn up(&self, transaction: &Transaction) -> Result<(), WalletMigrationError> {
+        transaction.execute_batch(
+            "CREATE VIEW IF NOT EXISTS v_tx_sent_notes AS
+            SELECT transactions.id_tx         AS id,
+                   transactions.block         AS mined_height,
+                   transactions.tx_index      AS transaction_index,
+                   transactions.txid          AS raw_transaction_id,
+                   transactions.expiry_height AS expiry_height,
+                   transactions.raw           AS raw,
+                   sent_notes.address         AS to_address,
+                   sent_notes.value           AS value,
+                   sent_notes.memo            AS memo,
+                   sent_notes.id_note         AS note_id,
+                   blocks.time                AS block_time
+            FROM   transactions
+                   LEFT JOIN sent_notes
+                          ON transactions.id_tx = sent_notes.tx
+                   LEFT JOIN blocks
+                          ON transactions.block = blocks.height;
+            CREATE VIEW IF NOT EXISTS v_tx_received_notes AS
+            SELECT transactions.id_tx     AS id,
+                   transactions.block     AS mined_height,
+                   transactions.tx_index  AS transaction_index,
+                   transactions.txid      AS raw_transaction_id,
+                   received_notes.value   AS value,
+                   received_notes.memo    AS memo,
+                   received_notes.id_note AS note_id,
+                   blocks.time            AS block_time,
+                   IFNULL(received_notes.is_change, false) as is_change
```

Comment:
Semantically, "is this note change" cannot be answered for a non-received note. So this really _should_ be null. It would be non-null if the view used an inner join instead (excluding transactions with no received notes).